### ISSUE TITLE
Braces and allman not applied to multi-line functions

### ISSUE THREAD

**Issue by jlnbuiles** [nicklockwood/SwiftFormat#1146]
I noticed that the `allman` formatting is not being applied to multi-line functions. e.g.:

For this guy **it works beautifully** 🌸:
```
func httpURLResponse(withResponse response: URLResponse) throws -> Bool {
    ...
}
```

But for this guy **it doesn't**:
```
func application(_: UIApplication,
                     didFinishLaunchingWithOptions _: [UIApplication.LaunchOptionsKey: Any]?) -> Bool
{// i wasn't taken up next to the return argumen
        ...
}
```

Is this intended behavior? if so maybe we should mention it in the docs? how could I accomplish consistent opening brace formatting for all methods?

What i'm setting on the config file:

```
--enable braces
...
--allman false
```

_Attaching the complete config file for reference as well. It could be that I have conflicting configuration._
[swiftformat 2.txt](https://github.com/nicklockwood/SwiftFormat/files/8133093/swiftformat.2.txt)

---
**Versions for all the stuffs**
- SwiftFormat version: 0.49.4
- Xcode version: 13.2.1
- Swift version: 5.5.2
_

**Comment by jlnbuiles** [comment]
also noticed it on a guard statement.

I had this:
```
guard let cell = tableView.dequeueReusableCell(withIdentifier: CryptoListsTableViewCell.identifier, for: indexPath) as? CryptoListsTableViewCell else {fatalError("Unable to create cell.")}
```        

and it was formatted tot this:
```
guard let cell = tableView.dequeueReusableCell(withIdentifier: CryptoListsTableViewCell.identifier,
                                                                          for: indexPath) as? CryptoListsTableViewCell else
{
    fatalError("Unable to create cell.")
}
```

**Comment by nicklockwood** [comment]
@jlnbuiles this behavior is causes by the `wrapMultilineStatementBraces` rule. You can disable it as follows:

```
--disable wrapMultilineStatementBraces
```

**Comment by jlnbuiles** [comment]
So i went ahead and did this and it does implement it in some cases, but not in others.

Worked for this one:
**initial format**
```
        guard let cell = tableView.dequeueReusableCell(withIdentifier: CryptoListsTableViewCell.identifier,
                                                       for: indexPath) as? CryptoListsTableViewCell else 
        {
            fatalError("Unable to create cell.")
        }
```

**resulting format**
```
        guard let cell = tableView.dequeueReusableCell(withIdentifier: CryptoListsTableViewCell.identifier,
                                                       for: indexPath) as? CryptoListsTableViewCell else {
            fatalError("Unable to create cell.")
        }
```

For these 2 it did not apply applied any formatting and left them as is:
```
    func application(_: UIApplication,
                     didFinishLaunchingWithOptions _: [UIApplication.LaunchOptionsKey: Any]?) -> Bool
    {
        // Override point for customization after application launch.
        return true
    }
```

and this one:
```
    func application(_: UIApplication,
                     configurationForConnecting connectingSceneSession: UISceneSession,
                     options _: UIScene.ConnectionOptions) -> UISceneConfiguration
    {
        // Called when a new scene session is being created.
        // Use this method to select a configuration to create the new scene with.
        return UISceneConfiguration(name: "Default Configuration", sessionRole: connectingSceneSession.role)
    }
```

¿is there a specific reason why that is?

Additionally, maybe it'd be helpful to add a small section within the [rules md](https://github.com/nicklockwood/SwiftFormat/blob/master/Rules.md#wrapMultilineStatementBraces) that lists explicitly rules that are NOT applied by default? it seems like most **are enabled** by default.

This would probably lead people to simpler configuration `.swiftformat` files. As they should only list rules they want to disable, or modify the default behavior or.

**Comment by nicklockwood** [comment]
@jlnbuiles fixed in 0.49.5

**Comment by jlnbuiles** [comment]
dopeness! thanks brother!
